tools/cold-store/src/cli.rs

```diff
@@ -143,6 +160,71 @@ fn copy_next_block(store: &NodeStorage, config: &NearConfig, hot_runtime: &Arc<N
         .expect(&std::format!("Failed to update cold HEAD to {}", next_height));
 }
 
+fn copy_all_blocks(store: &NodeStorage, batch_size: usize, check: bool) {
+    // If FINAL_HEAD is not set for hot storage though, we default it to 0.
+    let hot_final_head = store
+        .get_store(Temperature::Hot)
+        .get_ser::<Tip>(DBCol::BlockMisc, FINAL_HEAD_KEY)
+        .unwrap_or_else(|e| panic!("Error reading hot FINAL_HEAD: {:#}", e))
+        .map(|t| t.height)
+        .unwrap_or(0);
+
+    copy_all_data_to_cold(
+        (*store.cold_db().unwrap()).clone(),
+        &store.get_store(Temperature::Hot),
+        batch_size,
+    )
+    .expect("Failed to do migration to cold db");
+
+    update_cold_head(
+        &*store.cold_db().unwrap(),
+        &store.get_store(Temperature::Hot),
+        &hot_final_head,
+    )
+    .expect(&std::format!("Failed to update cold HEAD to {}", hot_final_head));
+
+    if check {
+        for col in DBCol::iter() {
+            if col.is_cold() {
+                println!(
+                    "Performed {} {:?} checks",
+                    check_iter(
+                        &store.get_store(Temperature::Hot),
+                        &store.get_store(Temperature::Cold),
+                        col,
+                    ),
+                    col
+                );
+            }
+        }
+    }
+}
+
+fn check_key(
+    first_store: &near_store::Store,
+    second_store: &near_store::Store,
+    col: DBCol,
+    key: &[u8],
+) {
+    let first_res = first_store.get(col, key);
+    let second_res = second_store.get(col, key);
+
+    assert_eq!(first_res.unwrap(), second_res.unwrap());
+}
+
+fn check_iter(
+    first_store: &near_store::Store,
+    second_store: &near_store::Store,
+    col: DBCol,
+) -> u64 {
+    let mut num_checks = 0;
+    for (key, _value) in first_store.iter(col).map(Result::unwrap) {
+        check_key(first_store, second_store, col, &key);
+        num_checks += 1;
+    }
```

Comment:
This is asymmetrical in that it checks that first_store <= second_store - but there can be some keys in second_store that are not in first_store. This may be fine for our use case as long as it's used correctly and we check that hot_store <= cold_store. Can you either make this function symmetrical or make sure it's documented that is is asymmetric and that it's used correctly?